File: samples/sandbox/sandbox.js
require('../..')
const root  = require('path').resolve(__dirname, '.')
    , sandbox = require('bindings')({ module_root: root, bindings: 'sandbox' });

sandbox.greet();
console.log(sandbox.five());
console.log(sandbox.hello());
var a = 2, b = 3;
console.log(a + ' + ' + b + ' = ' + sandbox.add(a, b));
try {
  sandbox.add("foo");
} catch (ex) {
  console.log(ex);
}

var world = new sandbox.World();

File: samples/sandbox/sandbox.cpp
#include <iostream>
#include <n2o.hpp>

using namespace n2o;

void greet() { std::cerr << "hello world" << std::endl; }
int  five() { return 5; }
char const* hello() { return "hello"; }
int  add(int a, int b) { return a + b; }

struct world {};

N2O_ADD_ON(object exports) {
  exports["greet"] = function(greet);
  exports["five"]  = function(five);
  exports["hello"] = function(hello);
  exports["add"]   = function(add);

  exports["World"] = class_<world>("World");
}
